File: src/backend/Ecommerce.Api/Responses/CheckoutResponse.cs
namespace Ecommerce.Api.Responses;

public sealed class CheckoutResponse
{
    public int OrderId { get; init; }
    public decimal TotalAmount { get; init; }
    public DateTime CreatedAtUtc { get; init; }
    public IReadOnlyList<CheckoutItemResponse> Items { get; init; } = [];
    public string Message { get; init; } = string.Empty;
}

File: src/frontend/src/app/models/checkout/checkout-response.ts
export interface CheckoutResponseItem {
  productId: number;
  productName: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
}

export interface CheckoutResponse {
  orderId: number;
  totalAmount: number;
  createdAtUtc: string;
  items: CheckoutResponseItem[];
  message: string;
}
